frontend/src/MainBlock/Repositories/FetchingParams/SortingAttributes.js

```diff
@@ -0,0 +1,91 @@
+import React, { Fragment, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import SortingAttribute from './SortingAttribute';
+
+const availableAttributes = {
+    name: {
+        label: 'Repository name',
+        state: null,
+    },
+    createdAt: {
+        label: 'Creating date',
+        state: null,
+    },
+    updatedAt: {
+        label: 'Updating date',
+        state: null,
+    },
+    repositoryType: {
+        label: 'Repository type',
+        state: null,
+    },
+};
+
+const availableStates = [null, 'asc', 'desc'];
+
+const getNextState = currentState => availableStates[
+    (availableStates.indexOf(currentState) + 1) % availableStates.length
+];
+
+function SortingAttributes({ sortedAttributes, setSortedAttributes }) {
+    const [sortAttributes, setSortAttributes] = useState(availableAttributes);
+
+    const getAttribute = attributeName => ({
+        name: attributeName,
+        label: sortAttributes[attributeName].label,
+        state: sortAttributes[attributeName].state,
+    });
+
+    const rebuildSortOrder = (attributeName, newState) => {
+        const attributeSortIndex = sortedAttributes.findIndex(item => item.name === attributeName);
+        const newSortOrder = [...sortedAttributes];
+
+        if (newState && attributeSortIndex === -1) {
+            newSortOrder.push(getAttribute(attributeName));
+        }
+
+        if (attributeSortIndex !== -1) {
+            if (newState) {
+                // Сортировка есть, атрибут есть - обновляем
+                newSortOrder[attributeSortIndex] = getAttribute(attributeName);
+            } else {
+                newSortOrder.splice(attributeSortIndex, 1);
+            }
+        }
+
+        setSortedAttributes(newSortOrder);
+    };
+
+    const handleClick = (attributeName) => {
+        if (!Object.prototype.hasOwnProperty.call(sortAttributes, attributeName)) {
```

Comment:
```suggestion
        if (!sortAttributes[attributeName]) {
```
Это IDEA так посоветовала?

Reply:
Исправила